ibis/backends/dask/client.py

```diff
@@ -0,0 +1,468 @@
+
+    pairs = []
+    for column_name, dask_dtype in df.dtypes.iteritems():
+        if not isinstance(column_name, str):
+            raise TypeError(
+                'Column names must be strings to use the dask backend'
+            )
+
+        if column_name in schema:
+            ibis_dtype = dt.dtype(schema[column_name])
+        elif dask_dtype == np.object_:
+            inferred_dtype = infer_dask_dtype(df[column_name], skipna=True)
+            if inferred_dtype in {'mixed', 'decimal'}:
+                # TODO: in principal we can handle decimal (added in pandas
+                # 0.23)
+                raise TypeError(
+                    'Unable to infer type of column {0!r}. Try instantiating '
+                    'your table from the client with client.table('
+                    "'my_table', schema={{{0!r}: <explicit type>}})".format(
+                        column_name
+                    )
+                )
+            ibis_dtype = _inferable_dask_dtypes[inferred_dtype]
+        else:
+            ibis_dtype = dt.dtype(dask_dtype)
+
+        pairs.append((column_name, ibis_dtype))
+
+    return sch.schema(pairs)
+
+
+def ibis_dtype_to_dask(ibis_dtype):
+    """Convert ibis dtype to the dask / numpy alternative"""
+    assert isinstance(ibis_dtype, dt.DataType)
+
+    if isinstance(ibis_dtype, dt.Timestamp) and ibis_dtype.timezone:
+        return DatetimeTZDtype('ns', ibis_dtype.timezone)
+    elif isinstance(ibis_dtype, dt.Interval):
+        return np.dtype('timedelta64[{}]'.format(ibis_dtype.unit))
+    elif isinstance(ibis_dtype, dt.Category):
+        return CategoricalDtype()
+    elif type(ibis_dtype) in _ibis_dtypes:
+        return _ibis_dtypes[type(ibis_dtype)]
+    else:
+        return np.dtype(np.object_)
+
+
+def ibis_schema_to_dask(schema):
+    return list(zip(schema.names, map(ibis_dtype_to_dask, schema.types)))
+
+
+convert = Dispatcher(
+    'convert',
+    doc="""\
+Convert `column` to the dask dtype corresponding to `out_dtype`, where the
+dtype of `column` is `in_dtype`.
+
+Parameters
+----------
+in_dtype : Union[np.dtype, dask_dtype]
+    The dtype of `column`, used for dispatching
+out_dtype : ibis.expr.datatypes.DataType
+    The requested ibis type of the output
+column : dd.Series
+    The column to convert
+
+Returns
+-------
+result : dd.Series
+    The converted column
+""",
+)
+
+
+@convert.register(DatetimeTZDtype, dt.Timestamp, dd.Series)
+def convert_datetimetz_to_timestamp(in_dtype, out_dtype, column):
+    output_timezone = out_dtype.timezone
+    if output_timezone is not None:
+        return column.dt.tz_convert(output_timezone)
+    return column.astype(out_dtype.to_dask(), errors='ignore')
+
+
+def convert_timezone(obj, timezone):
+    """Convert `obj` to the timezone `timezone`.
+
+    Parameters
+    ----------
+    obj : datetime.date or datetime.datetime
+
+    Returns
+    -------
+    type(obj)
+    """
+    if timezone is None:
```

Comment:
hmm, where did this come from? we really want to let Timestamp handle this and not do these types of conversions directly.

Reply:
Similarly: https://github.com/ibis-project/ibis/blob/master/ibis/backends/pandas/client.py#L241

Reply:
ok ic, yeah you can likely just import things like this